server/pkg/property/value_spaceing.go

```diff
@@ -0,0 +1,94 @@
+package property
+
+import (
+	"github.com/mitchellh/mapstructure"
+)
+
+var ValueTypeSpacing = ValueType("spacing")
+
+type Spacing struct {
+	Top    float64 `json:"top" mapstructure:"top"`
+	Bottom float64 `json:"bottom" mapstructure:"bottom"`
+	Left   float64 `json:"left" mapstructure:"left"`
+	Right  float64 `json:"right" mapstructure:"right"`
+}
+
+func (c *Spacing) Clone() *Spacing {
+	if c == nil {
+		return nil
+	}
+	return &Spacing{
+		Top:    c.Top,
+		Bottom: c.Bottom,
+		Left:   c.Left,
+		Right:  c.Right,
+	}
+}
+
+type typePropertySpacing struct{}
+
+func (*typePropertySpacing) I2V(i interface{}) (interface{}, bool) {
+	if v, ok := i.(Spacing); ok {
+		return v, true
+	}
+
+	if v, ok := i.(*Spacing); ok {
+		if v != nil {
+			return *v, true
+		}
+		return nil, false
+	}
+
+	v := Spacing{}
+	if err := mapstructure.Decode(i, &v); err == nil {
+		return v, true
+	}
+	return nil, false
+}
+
+func (*typePropertySpacing) V2I(v interface{}) (interface{}, bool) {
```

Comment:
The `V2I` method seems to not be performing any useful operation.

Reply:
this method exists because I'm implementing the `TypeProperty` interface and I'm forced to implement this method.